dashboard/state_aggregator.py

```diff
@@ -195,6 +197,35 @@ async def list_objects(self, *, option: ListApiOptions) -> dict:
         result.sort(key=lambda entry: entry["object_id"])
         return {d["object_id"]: d for d in islice(result, option.limit)}
 
+    async def list_runtime_envs(self, *, option: ListApiOptions) -> List[dict]:
```

Comment:
Can we use a structured type instead of `List[dict]` here?

Reply:
Do you suggest using something like Dataclass? Or is there a good way to represent type when the returned dict contains multiple types? Should I use sth like `List[Dict[str, Union[str, int]]`?

Reply:
dataclass would be ideal

Reply:
```
A list of runtime env information in the cluster. The schema of returned "dict"
            is equivalent to the `RuntimeEnvState` protobuf message.
```

I will write this comment for now. I think we need dataclasses for all APis here, not just this one (could be in a separate independent PR).

One concern is that using dataclass here requires pretty complicated conversion. For example, currently we are doing protobuf -> dict conversion, and that's it. But if we use dataclasses, it needs to do protobuf -> dict -> dataclass -> dict conversion. Do you happen to know dict -> dataclass & dataclass -> dict conversion is cheap or not?